### PR TITLE
More improvements to dash-example.html

### PR DESCRIPTION
**Pull request by tnyeanderson**

### CODE REVIEW

**Comment by andrewm4894** [comment]
Hey - this looks interesting, i'm a bit out of the loop but curious. I see another user on discord who maybe might actually be asking for something potentially similar to this. 

Do you have any example screenshots or some ELI5 type intro into exactly what this gives you? Maybe something super obvious that we could just add after the 1st paragraph in the readme.

I might try and figure this out and play with it myself over the break as am curious.

**Comment by andrewm4894** [comment]
Sweet! got it working myself - adding image here that i will link to in the README to make it super obvious to anyone interested. 

image

Might add one or two little small notes to the readme to make it as easy as possible for less technical people (like me :) ) to get it working.

**Comment by andrewm4894** [comment]
Love this, let me know if anything i can do to help. I'm ML guy so don't really know HTML or javascript or anything but happy to help test or anything. 

Would be nice if there was an option to hide the gridlines (white ones below sort of make it harder for my eyes) and/or overall dark theme vs light theme (not sure if that's easy in the charts api or not).

image

**Comment by andrewm4894** [comment]
being able to flip the layout between horizontal (one row per host) and vertical (one column per host) would be cool too. 

ok, apologies for the feature requests :)

**Comment by tnyeanderson** [comment]
First, let me say I'm so happy you are enjoying the dash page! It has been a lot of fun for me to make and I use it every day. Also happy to see you were able to figure out how to set it up, even with the room for improvement in the documentation. Can I ask if you are using the version from this PR or from the main branch?

>Would be nice if there was an option to hide the gridlines (white ones below sort of make it harder for my eyes)

I can look into this. Are you wanting to hide the vertical, horizontal, or both?

>overall dark theme vs light theme (not sure if that's easy in the charts api or not).

I'll have to check regarding the charts themselves. The rest of the application is pretty easy to theme (in this PR) using the first `<style>` block which contains the color variables used. Perhaps we could create a default "light theme" CSS block and have it commented out, then one could easily switch between the "themes" by commenting out one and uncommenting the other (**EDIT: this is a variable now**). Not sure how I feel about allowing the change on the fly as it would add complexity to the application (would have to change the graphs, etc as well) but I could be convinced (**EDIT: on the fly is too complex as it requires rewriting the graph/chart elements, meaning at best theme preference could be saved in local storage and the page reloaded, which I am not a fan of at this time... though again I could be convinced**). I would never use a light theme myself, which is why I didn't build one in, but I did try to leave the door open for theming by end users :)

>Might add one or two little small notes to the readme to make it as easy as possible for less technical people (like me :) ) to get it working.

I'm always looking for ways to improve documentation. It's hard to write docs for your own software because you know it too well to understand where people might be confused! That said, with this little rewrite there are definitely some docs that will need updated, just haven't gotten around to it yet. I have many many projects going on at the moment (not even counting my day job) and this one is unfortunately more of a fun, low-priority, "it works well enough" kind of thing for me :)

>being able to flip the layout between horizontal (one row per host) and vertical (one column per host) would be cool too.

I don't think this will be supported. The reason columns are used is because you can fit more hosts on the screen (as standard screens have more width than they do height). If we arranged the hosts in rows, you could only see two or *maybe* three at a time, which kind of defeats the purpose in my opinion.

It would also add too much complexity. I've tried to make the page flexible and simple. It is just over 1000 lines (~300 of those just devoted to being able to sort hosts) and to me that is clean and manageable as a single file. The more pieces that are added, the more tempted I would be to separate the concerns to different "module" files, which I don't want to do...

If you are trying to be able to see more stats at once, I can recommend zooming out to see more charts per host, *and* more hosts!

EDIT: I will also say that the existing column layout is due only to CSS. The alarms sidebar is a little more hardcoded in terms of styling, but really all that happens for the columns is that the host template element get copied for each host. So if you want to rearrange and restyle your host template, you can make it work any way you want. And if you do something cool, definitely submit it back to this community repo! :)

**Comment by tnyeanderson** [comment]
@andrewm4894 thanks again for your notes. There is now a `THEME` configuration option available that can be set to `dark` (default) or `light`. I've created a basic light theme based on the light theme already present on existing netdata dashboards, but I am definitely open to feedback on it.

I still need to update the docs on all this for the PR, I'm just being lazy. I appreciate your patience :)

**Comment by tnyeanderson** [comment]
Also found out how to hide the gridlines (at least with dygraph)... for instance:
```html
<div class="dash-graph"
    data-dash-netdata="system.load"
    data-dygraph-drawgrid="false">
</div>
```

Just add that `data-dygraph-drawgrid` attribute to any graphs you want to hide the lines on... Hope this helps :)

[relevant source code](https://github.com/netdata/netdata/blob/5b57fc441c40959514c4e2d0863be2e6a417e352/web/gui/dashboard.js#L2100)

**Comment by andrewm4894** [comment]
@tnyeanderson shall i merge this? I think yes but just wanted to double check.

**Comment by tnyeanderson** [comment]
Sorry about that. This totally fell off the radar for me. I have no further changes at the moment, can always create a new PR if needed :) 

Feel free to merge!

EDIT: Hmmm, reading back it looks like some docs updates were in order for this that I never did... If you are willing to wait a second I can try to get to it this weekend. Otherwise if you want to merge I can open a new PR for the docs update. Up to you

**Comment by andrewm4894** [comment]
cool - i don't mind waiting. Just trying to do some clean up on this repo and want to try get this merged in next few weeks maybe as think will be useful for others for sure.

**Comment by tnyeanderson** [comment]
Okay, I went ahead and did a last docs cleanup. I saw that you added some documentation yourself in the meantime--thank you so much for that!!

I'm good to merge this if you are.

And I hate to say, but this may be my last rodeo with this project. My homelab is migrating to prometheus to better align with what we are using at work (which, incidentally, is also ML related lol). Thanks again and sorry this one took me so long!!

**Comment by andrewm4894** [comment]
For some reason when i try use it i'm getting just this screen:

image

wonder if something has changed in meantime on agent side that has broken things...

**Comment by andrewm4894** [comment]
@tnyeanderson cool i actually was just playing with our Prometheus collector a little while back too. 

https://github.com/netdata/go.d.plugin/tree/master/modules/prometheus#configuration

Is an example config in there showing to pull from this demo prom node exporter endpoint : https://node.demo.do.prometheus.io/metrics

So could be easy enough to have both potentially. 

We are actually just working on [extending the ML](https://github.com/netdata/netdata/discussions/12763) based anomaly detection stuff at the moment.

Is the ML stuff anything related to ML based observability or more so it's just using Prometheus for observability stack relating to some other independent ML infra and stuff?

**Comment by tnyeanderson** [comment]
> For some reason when i try use it i'm getting just this screen:
>
> wonder if something has changed in meantime on agent side that has broken things...

Just to check, you did change the `NETDATA_HOST` when you copied it over? Nothing really changed in the last few commits, and I did just test it on my end and verified it works... perhaps you were relying on the `localhost` default, which has changed to be more documentative?

>Is the ML stuff anything related to ML based observability or more so it's just using Prometheus for observability stack relating to some other independent ML infra and stuff?

The latter, specifically for monitoring batch- and kubernetes-based ML platforms :)
